### PR TITLE
do not return empty event lists

### LINKED ISSUE #103: Consider removing the empty event lists returned by most function calls.
Initially we were being very defensive about the possibility that the user-level function calls (e.g. `send_headers` and friends) might want to return events. None of them do, and it seems really unlikely that any will. We should consider abandoning that pattern.

### PR DESCRIPTION
**Pull request by Kriechi**
fixes #103

### CODE REVIEW

File: h2/stream.py [hunk 1]
```diff
@@ -572,10 +572,9 @@ def locally_pushed(self):
         immediately after initialization. Sends no frames, simply updates the
         state machine.
         """
-        events = self.state_machine.process_input(
-            StreamInputs.SEND_PUSH_PROMISE
-        )
-        return [], events
+        # This does not trigger any events.
+        self.state_machine.process_input(StreamInputs.SEND_PUSH_PROMISE)
```
Comment (Lukasa) [inline h2/stream.py]: For any call into the state machine, can we assert on the return value? This'll stop us getting this wrong or accidentally throwing away anything we need.

**Comment by Lukasa** [comment]
This mostly looks reasonable. Just one note. =)

File: h2/stream.py (hunk 1 above)
Comment (Kriechi) [inline h2/stream.py]: better now?

File: h2/stream.py [hunk 2]
```diff
@@ -534,7 +534,13 @@ def send_headers(self, headers, encoder, end_stream=False):
         """
         # Because encoding headers makes an irreversible change to the header
         # compression context, we make the state transition *first*.
-        events = self.state_machine.process_input(StreamInputs.SEND_HEADERS)
+
+        # This does not trigger any events.
+        events = self.state_machine.process_input(
+            StreamInputs.SEND_HEADERS
+        )
+        assert len(events) == 0
```
Comment (Lukasa) [inline h2/stream.py]: Let's keep this idiomatic: `assert not events`.

**Comment by Lukasa** [comment]
Better! One small style change. Also, the doc strings might be better if rather than deleting the `:returns:` stanza we changed it to ’:returns: Nothing`. Thoughts?

**Comment by Lukasa** [comment]
Looks great, thanks!

**Pull request merged**